plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/wizards/FileManagementUtil.java

```diff
@@ -0,0 +1,399 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.wso2.developerstudio.humantaskeditor.Activator;
+import org.wso2.developerstudio.humantaskeditor.HumantaskEditorConstants;
+
+public class FileManagementUtil {
+    private static final Logger logger = Logger.getLogger(Activator.PLUGIN_ID);
+
+    public static void copyDirectory(File srcPath, File dstPath, List filesToBeCopied) throws IOException {
+
+        if (srcPath.isDirectory()) {
+            if (!dstPath.exists()) {
+                dstPath.mkdir();
+            }
+            String files[] = srcPath.list();
+            for (int i = 0; i < files.length; i++) {
+                copyDirectory(new File(srcPath, files[i]), new File(dstPath, files[i]), filesToBeCopied);
+            }
+        } else {
+            if (!filesToBeCopied.contains(srcPath.getAbsolutePath()))
+                return;
+            if (!srcPath.exists()) {
+                return;
+            } else {
+                FileManagementUtil.copy(srcPath, dstPath);
+            }
+        }
+    }
+
+    public static List getAllFilesPresentInFolder(File srcPath) {
+        List fileList = new ArrayList();
+        if (srcPath.isDirectory()) {
+            String files[] = srcPath.list();
+            for (int i = 0; i < files.length; i++) {
+                fileList.addAll(getAllFilesPresentInFolder(new File(srcPath, files[i])));
+            }
+        } else {
+            fileList.add(srcPath.getAbsolutePath());
+        }
+        return fileList;
+    }
+
+    public static void removeEmptyDirectories(File srcPath) {
+        if (srcPath.isDirectory()) {
+            String files[] = srcPath.list();
+            for (int i = 0; i < files.length; i++) {
+                removeEmptyDirectories(new File(srcPath, files[i]));
+            }
+            if (srcPath.list().length == 0) {
+                srcPath.delete();
+            }
+        }
+    }
+
+    static public void zipFolder(String srcFolder, String destZipFile) throws Exception {
+        ZipOutputStream zip = null;
+        FileOutputStream fileWriter = null;
+        try {
+            fileWriter = new FileOutputStream(destZipFile);
+            zip = new ZipOutputStream(fileWriter);
+        } catch (FileNotFoundException e) {
+            logger.log(Level.FINE, HumantaskEditorConstants.ERROR_CREATING_CORRESPONDING_ZIP_FILE, e);
+        }
+        addFolderContentsToZip(srcFolder, zip);
+        try {
+            zip.flush();
+            zip.close();
+        } catch (IOException ex) {
+            logger.log(Level.FINE, HumantaskEditorConstants.ERROR_CREATING_CORRESPONDING_ZIP_FILE, ex);
+        }
+    }
+
+    static private void addToZip(String path, String srcFile, ZipOutputStream zip) {
```

Comment:
You can find the implementation of this method in some other places too. If found use it other that writing again. Else make this a common method and send a PR to tooling platform.